models/Element/Tag.php

```diff
@@ -70,14 +72,21 @@ final class Tag extends Model\AbstractModel
      */
     public static function getById(int $id): ?Tag
     {
-        try {
-            $tag = new self();
-            $tag->getDao()->getById($id);
+        $cacheKey = 'tags_' . $id;
 
-            return $tag;
-        } catch (Model\Exception\NotFoundException $e) {
-            return null;
+        try {
+            $tag = RuntimeCache::get($cacheKey);
+        } catch (\Exception $ex) {
+            try {
+                $tag = new self();
+                $tag->getDao()->getById($id);
+                RuntimeCache::set($cacheKey, $tag);
```

Comment:
@kingjia90 RuntimeCache should be cleared in the `delete()` method, right?

Reply:
Thank you for reviewing!

I've tweaked the DAO `delete()` to fetch all the ids to be cleared, so i can clear the runtimeCache of all the deleted tags, or do you know if there is a better way by any chance?

Reply:
@kingjia90 I looked in the AbstractObject delete() method. There it is iterated through the child elements and the delete method is executed again. The runtime cache could then remain in the model.

But then all child elements must be loaded. I think this will be much slower in terms of performance.

Reply:
Thank you for the feedback, since it's a final class (model) and internal (dao), it would be just safe (and more cleaner) to do like this https://github.com/pimcore/pimcore/pull/17390/commits/9cf5a7a59b331c7661e454c83fe9d5ff8f82a45e

wdyt?

Reply:
@blankse could you have a quick confirmation for this? Thank you in advance

Reply:
mhhh Blankse must be on a well deserved vacation 😄 merging it, in case we open a follow up